File: BilDunya/src/main/java/com/bildunya/dto/CreateCustomLocationRequest.java
package com.bildunya.dto;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class CreateCustomLocationRequest {

    @NotBlank(message = "Name is required")
    private String name;

    private String description;

    @NotNull(message = "Latitude is required")
    private Double latitude;

    @NotNull(message = "Longitude is required")
    private Double longitude;

    private List<String> tags;
}

File: bildunya_frontend/lib/data/models/create_custom_location_request.dart
class CreateCustomLocationRequest {
  const CreateCustomLocationRequest({
    required this.name,
    this.description,
    required this.latitude,
    required this.longitude,
    this.tags,
  });

  final String name;
  final String? description;
  final double latitude;
  final double longitude;
  final List<String>? tags;

  Map<String, dynamic> toJson() => <String, dynamic>{
        'name': name,
        'description': description,
        'latitude': latitude,
        'longitude': longitude,
        'tags': tags,
      };
}

